plugins/pastetools/filter/image.js

```diff
@@ -97,8 +252,77 @@
 		return ret;
 	}
 
+	function getImageType( imageContent ) {
+		var tests = [
+			{
+				marker: /\\pngblip/,
+				type: 'image/png'
+			},
+
+			{
+				marker: /\\jpegblip/,
+				type: 'image/jpeg'
+			},
+
+			{
+				marker: /\\emfblip/,
+				type: 'image/emf'
+			},
+
+			{
+				marker: /\\wmetafile\d/,
+				type: 'image/wmf'
+			}
+		],
+		extractedType = CKEDITOR.tools.array.find( tests, function( test ) {
+			return test.marker.test( imageContent );
+		} );
+
+		if ( extractedType ) {
+			return extractedType.type;
+		}
+
+		return 'unknown';
+	}
+
 	function createSrcWithBase64( img ) {
+		var isSupportedType = CKEDITOR.tools.array.indexOf( CKEDITOR.pasteFilters.image.supportedImageTypes, img.type ) !== -1;
+
+		if ( !isSupportedType ) {
+			return null;
+		}
+
 		return img.type ? 'data:' + img.type + ';base64,' + CKEDITOR.tools.convertBytesToBase64( CKEDITOR.tools.convertHexStringToBytes( img.hex ) ) : null;
 	}
-
 } )();
+
+/**
+ * Virtual class that illustrates image data
+ * returned by {@link CKEDITOR.plugins.pastetools.filters.image#extractFromRtf} method.
+ *
+ * @since 4.16.0
+ * @class CKEDITOR.plugins.pastetools.filters.image.ImageData
+ * @abstract
+ */
+
+/**
+ * Unique id of an image extracted from RTF content.
```

Comment:
How this uniqueness relates to duplicated images - https://github.com/ckeditor/ckeditor4/pull/4291/files#diff-b2904cf407e5f480ff0c92f186507741R171. Would hex and type be identical too in such cases?

Reply:
Yes, they're the same images after all, so it makes _some_ sense.